src/libraries/System.Diagnostics.Process/tests/ProcessModuleTests.cs

```diff
@@ -89,5 +93,33 @@ public void ModulesAreDisposedWhenProcessIsDisposed()
             process.Dispose();
             Assert.Equal(expectedCount, disposedCount);
         }
+
+        [ActiveIssue("https://github.com/dotnet/runtime/pull/335059")]
+        [ConditionalFact(nameof(AreAllLongPathsAvailable))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void LongModuleFileNamesAreSupported()
+        {
+            // To be able to test Long Path support for ProcessModule.FileName we need a .dll that has a path >= 260 chars.
+            // Since Long Paths support can be disabled (see the ConditionalFact attribute usage above),
+            // we just copy "LongName.dll" from bin to a temp directory with a long name and load it from there.
+            // Loading from new path is possible because the type exposed by the assembly is not referenced in any explicit way.
+            const string libraryName = "LongName.dll";
+
+            string testBinPath = Path.GetDirectoryName(typeof(ProcessModuleTests).Assembly.Location);
+            string libraryToCopy = Path.Combine(testBinPath, libraryName);
+            Assert.True(File.Exists(libraryToCopy), $"{libraryName} was not present in bin folder '{testBinPath}'");
+
+            string directoryWithLongName = Path.Combine(TestDirectory, new string('a', Math.Max(1, 261 - TestDirectory.Length)));
```

Comment:
If `TestDirectory.Length` equals any number larger than 260, then `directoryWithLongName` will end up being `a`. Is that the intention?

Reply:
Yes, the intention here is to ensure that `directoryWithLongName` is longer than 260 characters. In this particular case, it would be `TestDirectory/a` with total length > 260